src/Microsoft.TemplateEngine.Abstractions/Parameters/DataSource.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Abstractions.Parameters
+{
+    public enum DataSource
+    {
+        /// <summary>
+        /// Value was not set.
+        /// </summary>
+        NoSource,
+
+        /// <summary>
+        /// Those are values supplied by the host. This usually means value(s) was/were set by user.
+        /// </summary>
+        Host,
```

Comment:
I was bit split on this - as value is simply passed by host (where it can possibly come from different origin than being sourced from user).
Or I can rename it to `User` and add something like `HostInternal` or `HostOther` for the custom hosts to indicate value is passed to template engine but actually not comming from user.
Thoughts?